kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -760,75 +760,99 @@ define([
                 })
                 .off('click');
         },
-  
+
         enableImportButton: function() {
-            let stagingAreaViewer = this; 
-  
+            let stagingAreaViewer = this;
+
             this.$elem.find('button.kb-staging-table-import__button')
                 .removeClass('kb-staging-table-import__button__disabled')
                 .tooltip('disable')
                 .off('click')
                 .on('click keyPress', function() {
-                    stagingAreaViewer.initImport();
+                    stagingAreaViewer.initBulkImport();
                 });
         },
-  
-        initImport: function() {
-            let stagingAreaViewer = this; 
-  
-            //get all of the selected checkbox file names and import type
+
+        /**
+         * Initializes the bulk import process. This takes the bulk-importer-supported types and
+         * builds a Bulk Import cell to address them. Other types have one import app cell
+         * generated for each.
+         *
+         * For example, if FASTQ files are supported by the bulk import cell, and you have
+         * 10 of those selected, and 2 genomes, then a single Bulk import cell will be created for
+         * the 10 FASTQ files, and 2 more cells are generated for each genome.
+         *
+         * If no files are selected by their checkbox, then no new cells will be created.
+         */
+        initBulkImport: function() {
+            const stagingAreaViewer = this;
+
+            // keys = types, values = list of files to be uploaded as that type
+            const bulkMapping = {};
+            // get all of the selected checkbox file names and import type
             $('input.kb-staging-table-body__checkbox-input:checked')
                 .each(function () {
                     const importType = $(this).attr('data-type');
                     const importFile = $(this).attr('data-file-name');
-                    stagingAreaViewer.initImportApp(importType, importFile);
+                    if (stagingAreaViewer.bulkImportTypes.includes(importType)) {
+                        if (!(importType in bulkMapping)) {
+                            bulkMapping[importType] = [];
+                        }
+                        bulkMapping[importType].push(importFile);
+                    }
+                    else {
+                        stagingAreaViewer.initImportApp(importType, importFile);
+                    }
                 });
-          
+            if (bulkMapping) {
```

Comment:
won't this always return true since we have it defined as an obj on line 791?

Reply:
Oops, yeah, probably. I meant to have something more descriptive there.

Reply:
Updated with a fix.